### ISSUE TITLE
Vue-Loader error when using reactivityTransform

### ISSUE THREAD

**Issue by craigrileyuk** [sxzz/unplugin-vue-macros#188]
### Describe the bug

Trying to set this up with Laravel Mix, Inertia and Vue2, but reactivityTransform isn't working.

```
ERROR in ./resources/js/pages/Home.vue?vue&type=template&id=b3c5cf30& (./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-vue-define-render!./node_modules/babel-loader/lib/index.js??clonedRuleSet-5.use[0]!./node_modules/vue-loader/lib/loaders/templateLoader.js??ruleSet[1].rules[14]!./node_modules/vue-loader/lib/index.js??vue-loader-options!./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-vue-define-options!./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-reactivity-transform-vue2!./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-vue-define-slots!./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-vue-define-model!./node_modules/unplugin/dist/webpack/loaders/transform.js?unpluginName=unplugin-vue-define-props!./resources/js/pages/Home.vue?vue&type=template&id=b3c5cf30&)
Module Error (from ./node_modules/unplugin/dist/webpack/loaders/transform.js):
unplugin-reactivity-transform-vue2 SyntaxError: This experimental syntax requires enabling one of the following parser plugin(s): "jsx", "flow", "typescript". (1:0)
```

```js
// webpack.mix.js
mix.webpackConfig({
    output: {
        chunkFilename: "js/[name].js?id=[chunkhash]",
    },
    plugins: [
        vueMacros({
            version: 2,
            defineOptions: {
                include: [/\.vue$/, /\.vue\?vue/],
            },
            reactivityTransformVue2: {
                include: [/\.vue$/, /\.vue\?vue/],
            },
        }),
    ],
});

mix.js("resources/js/app.js", "public/js").vue({ version: 2 });

mix.postCss("resources/css/app.css", "public/css");
```

### Reproduction

https://github.com/craigrileyuk/vue2-reactivity-transform

### System Info

```Shell
System:
    OS: macOS 12.6.2
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
    Memory: 1.54 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.18.0 - ~/.nvm/versions/node/v16.18.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.18.0/bin/yarn
    npm: 9.2.0 - ~/.nvm/versions/node/v16.18.0/bin/npm
  Browsers:
    Chrome: 108.0.5359.124
    Firefox: 108.0.1
    Safari: 16.2
```

### Used Package Manager

npm

### Validations

- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sxzz** [comment]
Thanks for your feedback. Fixed in `unplugin-vue-macros@1.2.1`

**Comment by sxzz** [comment]
You can also refer to [Vue 2 + Vue CLI example](https://github.com/vue-macros/vue2-vue-cli/commit/1d9c7994c4d57aa83fdf52ea635e6ea92336c6bd)
